constraints/src/main/java/gov/nasa/jpl/aerie/constraints/json/ConstraintsDSL.java

```diff
@@ -0,0 +1,68 @@
+package gov.nasa.jpl.aerie.constraints.json;
+
+import gov.nasa.jpl.aerie.constraints.model.Violation;
+import gov.nasa.jpl.aerie.constraints.time.Windows;
+import gov.nasa.jpl.aerie.constraints.tree.And;
+import gov.nasa.jpl.aerie.constraints.tree.Or;
+import gov.nasa.jpl.aerie.constraints.tree.Expression;
+import gov.nasa.jpl.aerie.constraints.tree.True;
+import gov.nasa.jpl.aerie.constraints.tree.ViolationsOf;
+import gov.nasa.jpl.aerie.json.Iso;
+import gov.nasa.jpl.aerie.json.JsonParser;
+import gov.nasa.jpl.aerie.json.Unit;
+
+import java.util.List;
+
+import static gov.nasa.jpl.aerie.json.BasicParsers.chooseP;
+import static gov.nasa.jpl.aerie.json.BasicParsers.listP;
+import static gov.nasa.jpl.aerie.json.BasicParsers.literalP;
+import static gov.nasa.jpl.aerie.json.BasicParsers.productP;
+import static gov.nasa.jpl.aerie.json.BasicParsers.recursiveP;
+import static gov.nasa.jpl.aerie.json.Uncurry.tuple;
+import static gov.nasa.jpl.aerie.json.Uncurry.untuple;
+
+public class ConstraintsDSL {
+
+  private static JsonParser<And> andF(final JsonParser<Expression<Windows>> windowsExpressionP) {
+    return productP
+        .field("kind", literalP("And"))
+        .field("expressions", listP(windowsExpressionP))
+        .map(Iso.of(untuple(($, expressions) -> new And(expressions)),
+                    // This parser is only used for deserialization, so we don't care about the reverse operation.
+                    $ -> tuple(Unit.UNIT, $.expressions)));
+  }
+
+  private static JsonParser<Or> orF(final JsonParser<Expression<Windows>> windowsExpressionP) {
```

Comment:
Hoping on this prematurely so no rush in responding.
I think this is supposed to be `orP` (same with `andF` above being a typo for `andP`).

Reply:
This nomenclature seems to go back to a PR by @kroffo [[AERIE-561] Deserialize constraint definitions](https://github.jpl.nasa.gov/Aerie/aerie/pull/732)

My _guess_ is the `F` ending stands for "Function"? These are all static methods that take other parsers and combine them in various ways.